syntax = "proto3";

package kvstore;

message PutRequest {
  bytes key = 1;
  bytes value = 2;
}

message PutReply {
  // Empty because success/failure is signaled via GRPC status.
}

message GetRequest {
  bytes key = 1;
}

message GetReply {
  bytes value = 1;
}

message RemoveRequest {
  bytes key = 1;
}

message RemoveReply {
  // Empty because success/failure is signaled via GRPC status.
}

message Kvpairs {
  repeated PutRequest pair = 1;
}

service KeyValueStore {
  rpc Put (PutRequest) returns (PutReply) {}
  rpc Remove (RemoveRequest) returns (RemoveReply) {}
  rpc Get (stream GetRequest) returns (stream GetReply) {}
}